src/packages/core/repository/repository-items.manager.ts

```diff
@@ -76,10 +87,25 @@ export class UmbRepositoryItemsManager<ItemType> extends UmbBaseController {
 		const { asObservable } = await this.repository.requestItems(this.getUniques());
 
 		if (asObservable) {
-			this.observe(asObservable(), (data) => this.#items.next(data), '_observeRequestedItems');
+			this.observe(
+				asObservable(),
+				(data) => {
+					this.#items.next(this.#sortByUniques(data));
+				},
+				'_observeRequestedItems',
+			);
 		}
 	}
 
+	#sortByUniques(data: Array<ItemType>) {
+		const uniques = this.getUniques();
+		return [...data].sort((a, b) => {
```

Comment:
Can't we use this here?

```
var uniqueAndSorted = [...new Set(myData)].sort() 
```

https://stackoverflow.com/a/42495707

Reply:
Thanks @bjarnef.  The intent of the `#sortByUniques` method is to sort the `data` array (of objects) in the order of the unique values. More like this: https://stackoverflow.com/q/13304543/12787

Reply:
Ahh okay, I haven't checked if `Set()` eventually can sort unique data objects in array:
https://stackoverflow.com/a/43667135